cobigen/cobigen-core/src/main/java/com/devonfw/cobigen/impl/config/ConfigurationFactory.java

```diff
@@ -0,0 +1,75 @@
+package com.devonfw.cobigen.impl.config;
+
+import java.net.URI;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Map;
+
+import com.devonfw.cobigen.api.util.MavenCoordinate;
+import com.devonfw.cobigen.impl.config.entity.Trigger;
+import com.devonfw.cobigen.impl.util.FileSystemUtil;
+import com.google.common.collect.Maps;
+
+/**
+ * The {@link ConfigurationFactory} creates one of the configuration readers fitting to the configuration xml
+ */
+public class ConfigurationFactory {
+
+  private Path configRoot;
```

Comment:
Shouldn't a Factory be stateless?

Reply:
We need the root in order to create the Configurations but since the interface structure was reverted, I'm not even sure if we still need the factory.